include/deal.II/sundials/kinsol.h

```diff
@@ -618,6 +618,30 @@ namespace SUNDIALS
      */
     std::function<VectorType &()> get_function_scaling;
 
+
+    /**
+     * A function object that users may supply and which is intended to perform
+     * custom setup on the supplied @p kinsol_mem object. Refer to the
+     * SUNDIALS documentation for valid options.
+     *
+     * For instance, the following code attaches a files for error output of the
+     * internal KINSOL implementation:
+     *
+     * @code
+     *      ode.custom_setup = [&](void *arkode_mem) {
+     *        KINSetErrFile(arkode_mem, errfile);
```

Comment:
Can you also show the code to set up `errfile`?

Separately, is `arkode_mem` really a `void *` or does ARKode use an actual type for that?

Reply:
Internally they likely use a custom type for that but to users this type is always erased. When calling `KINCreate` we are only handed a `void*` and are supposed to hand it back to all "member functions". It's a C Library :shrug:

Reply:
Ha, yes :-)